docs/frontend-development/frontend-build-process.md

```diff
@@ -0,0 +1,268 @@
+
+Before working with the frontend build process, ensure you have:
+
+- [Node.js](https://nodejs.org/) (version 16 or later)
+- [npm](https://www.npmjs.com/) (usually comes with Node.js)
+- Basic understanding of JavaScript, CSS, and build tools
+- Local development environment set up (see [Local Development](../testing-quality/local-development.md))
+
+## Build Tools
+
+The WP Scaffold project uses the following build tools:
+
+- **Webpack**: Module bundler for JavaScript and other assets
+- **Babel**: JavaScript compiler for using next-generation JavaScript features
+- **PostCSS**: Primary CSS processor with plugins for modern CSS features, transformations, and optimizations
+- **ESLint**: JavaScript linter for identifying and fixing code issues
+- **Stylelint**: CSS linter for consistent code style
+
+## Project Structure
+
+The frontend assets in the WP Scaffold themes are organized in the following structure:
+
+```
+wp-content/themes/[theme-name]/
+├── assets/
+│   ├── js/
+│   │   ├── frontend/
+│   │   │   ├── components/
+│   │   │   └── frontend.js
+│   │   └── block-editor/
+│   ├── css/
+│   │   ├── frontend/
+│   │   ├── blocks/
+│   │   ├── globals/
+│   │   └── mixins/
+│   ├── images/
+│   ├── fonts/
+│   └── svg/
+├── dist/ (compiled assets)
+│   ├── css/
+│   ├── js/
+│   └── images/
+├── blocks/ (Gutenberg blocks)
+├── src/ (PHP files)
+├── package.json
+└── webpack.config.js
+```
+
+## NPM Scripts
+
+The project includes several npm scripts to streamline the build process. These scripts are defined in the `package.json` file:
+
+```json
+{
+  "scripts": {
+    "start": "webpack --watch --mode=development",
+    "build": "webpack --mode=production",
+    "lint:js": "eslint assets/js/src",
+    "lint:css": "stylelint assets/css/**/*.css",
+    "lint": "npm run lint:js && npm run lint:css",
+    "test": "jest"
+  }
+}
+```
+
+### Common Commands
+
+- `npm start`: Start the development build process with file watching
+- `npm run build`: Create a production-ready build
+- `npm run lint`: Run linting on JavaScript and CSS files
+- `npm test`: Run JavaScript tests
+
+## Webpack Configuration
+
+The Webpack configuration is defined in `webpack.config.js`. Here's a simplified example of the configuration:
+
+```javascript
+const path = require('path');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+module.exports = (env, argv) => {
+  const isProduction = argv.mode === 'production';
+
+  return {
+    entry: {
+      main: './assets/js/src/main.js',
+      // Add additional entry points as needed
+    },
+    output: {
+      path: path.resolve(__dirname, 'assets/js/dist'),
+      filename: '[name].js',
+    },
+    devtool: isProduction ? false : 'source-map',
+    module: {
+      rules: [
+        // JavaScript
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          use: {
+            loader: 'babel-loader',
```

Comment:
Given that the project uses toolkit and doesn't have a custom webpack config I would remove this